iothub_client/inc/iothub_message.h

```diff
@@ -191,9 +191,14 @@ MOCKABLE_FUNCTION(, MAP_HANDLE, IoTHubMessage_Properties, IOTHUB_MESSAGE_HANDLE,
 *
 * @param   iotHubMessageHandle Handle to the message.
 *
-* @param   key name of the property to set.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
+* @param   key name of the property to set. Note that when sending messages via the HTTP transport, this value must not contain spaces.
 *
-* @param   value of the property to set.
+* @param   value of the property to set. 
+*
+*            @b NOTE: Generally, property key names and values can only contain ASCII alphanumeric characters, 
+*            plus {'!', '#', '$', '%, '&', ''', '*', '+', '-', '.', '^', '_', '`', '|', '~'}. However, the accepted  
+*            character sets are dependent on other factors such as the protocol being used. For more information
+*            on the character sets accepted by Azure IoT Hub, see <a href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct">Create and read IoT Hub messages</a>.
 *
```

Comment:
Each protocol is different enough that I would just point them to the doc.  So something like:

NOTE: The accepted character sets are dependent on different factors, such as the protocol being used. For more information on the character sets accepted ... (the rest you have)

Reply:
Ok, I trimmed it down as you recommended

Reply:
Thanks, looks good